packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -97,261 +105,282 @@ export const Groupings = ({ mapping, goBack }: GroupsTreeProps) => {
-              <>
-                {isLoadingQuery ? (
-                  value.row.original.groupName
-                ) : (
-                  <div
-                    className='iui-anchor'
-                    onClick={(e) => {
-                      e.stopPropagation();
-                      openProperties(value);
-                    }}
-                  >
-                    {value.row.original.groupName}
-                  </div>
-                )}
-              </>
-            ),
-          },
-          {
-            id: "description",
-            Header: "Description",
-            accessor: "description",
-          },
-          {
-            id: "dropdown",
-            Header: "",
-            width: 80,
-            Cell: (value: CellProps<GroupType>) => {
-              return (
-                <div onClick={(e) => e.stopPropagation()}>
-                  <DropdownMenu
-                    disabled={isLoadingQuery}
-                    menuItems={(close: () => void) => [
-                      <MenuItem
-                        key={0}
-                        onClick={() => onModify(value)}
-                        icon={<SvgEdit />}
-                      >
-                        Modify
-                      </MenuItem>,
-                      <MenuItem
-                        key={1}
-                        onClick={() => openProperties(value)}
-                        icon={<SvgList />}
-                      >
-                        Properties
-                      </MenuItem>,
-                      <MenuItem
-                        key={2}
-                        onClick={() => {
-                          setSelectedGroup(value.row.original);
-                          setShowDeleteModal(true);
-                          close();
-                        }}
-                        icon={<SvgDelete />}
-                      >
-                        Remove
-                      </MenuItem>,
-                    ]}
-                  >
-                    <IconButton
-                      disabled={isLoadingQuery}
-                      styleType='borderless'
-                    >
-                      <SvgMore
-                        style={{
-                          width: "16px",
-                          height: "16px",
-                        }}
-                      />
-                    </IconButton>
-                  </DropdownMenu>
-                </div>
-              );
-            },
-          },
-        ],
-      },
-    ],
-    [isLoadingQuery, onModify, openProperties],
+  const openProperties = useCallback(
+    async (group) => {
+      setSelectedGroup(group);
+      setGroupsView(GroupsView.PROPERTIES);
+      if (group && hiddenGroupsIds.includes(group.id)) {
+        await showGroup(group);
+        setHiddenGroupsIds(hiddenGroupsIds.filter((id) => id !== group.id));
+      }
+    },
+    [showGroup, hiddenGroupsIds, setHiddenGroupsIds],
   );
```

Comment:
All these functions don't require a useCallback. I don't see any dependency that actually needs them outside of themselves.